sieve/sievec.c

```diff
@@ -91,7 +91,7 @@ int main(int argc, char **argv)
         exit(1);
     }
 
-    instream = fopen(argv[optind++],"r");
+    instream = !strcmp(argv[optind], "-") ? stdin : fopen(argv[optind], "r");
     if(instream == NULL) {
         fprintf(stderr, "Unable to open %s for reading\n", argv[1]);
```

Comment:
Unrelated to this branch, but while you're in the area anyway, `argv[1]` in this error message should be `argv[optind]`

Reply:
Fixed in master and rebased